### ISSUE TITLE
Bug in DOCX: can not create hyperlink to figures correctly

### ISSUE THREAD

**Issue by Astro-Aionia** [lierdakil/pandoc-crossref#374]
Hi, I found hyperlinks to figures actually located to word bookmarks with the same name as fig ids defined in markdown files. But in fact, these bookmarks were not exist in the output DOCX document, so when I used the hyperlinks with <kbd>Ctrl</kbd>+<kbd>L Click</kbd>, they always linked to the start of the document.

I have referred #147 and tried  the lua script liedakil provided, but it didn't work. I'm wondering if there are some way ti fix the hyperlink for the latest pandoc-crossref.

**Comment by lierdakil** [comment]
Hi. Thanks for the report! Could you specify versions of pandoc-crossref and pandoc you're using, just to make sure we're on the same page?

I vaguely suspect this might be the same issue all over again, since Pandoc drastically changed how it represents figures in v3.0.

**Comment by Astro-Aionia** [comment]
I am using Pandoc 3.0 and pandoc-crossref 0.3.15.0.  Sorry for forgetting to illustrate before.

**Comment by lierdakil** [comment]
Okay, so I've taken a quick look, and it seems Pandoc does create bookmarks for divs, spans, code blocks, tables and plain images but indeed not figures. I should be able to whip up a PR over the weekend.

As a stopgap measure, this lua filter should work (it just wraps all figures in divs with the same identifier):
```lua
function Figure(el)
  return pandoc.Div(el, pandoc.Attr(el.identifier))
end
```

**Comment by Astro-Aionia** [comment]
Thanks for your reply. I have just tried this script. It works well for me.

**Comment by FeralFlora** [comment]
@lierdakil The issue was just fixed upstream. Not released yet, though.

**Comment by lierdakil** [comment]
Thanks for the update. I guess I'll need to remember to close this when updating to the next pandoc release... which all things considered is somewhat improbable, but I'll try :sweat_smile:

**Comment by FeralFlora** [comment]
I'll try to post a reminder at that time ;)

**Comment by lierdakil** [comment]
Presumably was fixed in pandoc v3.2.1, so a couple releases ago.
